CMakeLists.txt

```
@@ -19,6 +19,7 @@ option(RUN_LONG_UNIT_TESTS "Tests that takes more than 5 minutes to complete. No
 option(BUILD_STORAGE_SAMPLES "Build sample application for Azure Storage clients" OFF)
 option(BUILD_PERFORMANCE_TESTS "Build the performance test library" OFF)
 option(MSVC_USE_STATIC_CRT "(MSVC only) Set to ON to link SDK with static CRT (/MT or /MTd switch)." OFF)
+option(BUILTIN_ENV_LOGGER "Build suport for enabling logging to console when AZURE_LOG_LEVEL environment variable is set." ON)
```

We use `BUILD_XXX_XXX` for all other optional components.
Can you rename to `BUILD_AZURE_ENV_LOGGER` or `BUILD_ENV_LOGGER` ?

---

Following what other SDKs do, logging should be off be default.
```suggestion
option(BUILD_AZURE_CONSOLE_LOGGER "Build support for enabling logging to console when AZURE_LOG_LEVEL environment variable is set." OFF)
```

---

This is not about the logging being off, it is about having the code to listen to the environment variable `AZURE_LOG_LEVEL` on start up, and enabling console logging IF `AZURE_LOG_LEVEL` env variable is set in the environment.
Java has that functionality ON by default - https://github.com/Azure/azure-sdk-for-java/blob/5672b860b15aa32d1dd5b559f47c3cb1058f0166/sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java#L66 - in fact, I don't think you can even compile out that functionality from the Java SDK.

---

That 's a bit confusing. Do we really need this env var? Why don't we always look for `AZURE_LOG_LEVEL` and if it is set, we honor it/turn logging on.